sdk-setup/src/mb2

```diff
@@ -1853,7 +1856,7 @@ maybe_fix_package_version() {
         fatal "Cannot save state of Git submodules: git-stash failed for some."
     fi
 
-    if [[ $(git_ rev-parse "$tag^{}") != "$head" || $modified_submodules ]]; then
+    if [[ $head != "$tag_sha1" || $modified_submodules ]]; then
```

Comment:
Why is "$tag_sha1" quoted but "$head" is not?

Reply:
While word splitting and pathname expansion are not performed on the words between the `[[` and `]]`, so there is no need for quoting between these in general, a string to the right of any of the `==/!=/=~` operators is considered a pattern and matched, so (parts of) it needs to be quoted unless this is desired.